packages/vite-plugin-web-extension/src/plugins/hmr-rewrite-plugin.ts

```diff
@@ -61,32 +54,34 @@ export function hmrRewritePlugin(config: {
           // These are used in node_modules/vite/dist/client/client.mjs, check there to see if a var
           // can be null or not.
           __MODE__: JSON.stringify(config.mode || null),
-          __BASE__: JSON.stringify(server.base || "/"),
+          __BASE__: JSON.stringify(serverOptions.base || "/"),
           __DEFINES__: serializeDefine(config.define || {}),
-          __SERVER_HOST__: JSON.stringify(server.host || "localhost"),
-          __HMR_PROTOCOL__: JSON.stringify(hmr?.protocol || null),
-          __HMR_HOSTNAME__: JSON.stringify(hmr?.host || "localhost"),
-          __HMR_PORT__: JSON.stringify(hmr?.clientPort || hmr?.port || 5173),
+          __SERVER_HOST__: JSON.stringify(serverOptions.host || "localhost"),
+          __HMR_PROTOCOL__: JSON.stringify(hmrOptions?.protocol || null),
+          __HMR_HOSTNAME__: JSON.stringify(hmrOptions?.host || "localhost"),
+          __HMR_PORT__: JSON.stringify(
+            hmrOptions?.clientPort || hmrOptions?.port || 5173
+          ),
           __HMR_DIRECT_TARGET__: JSON.stringify(
-            `${server.host ?? "localhost"}:${server.port ?? 5173}${
-              config.base || "/"
-            }`
+            `${serverOptions.host ?? "localhost"}:${
+              serverOptions.port ?? 5173
+            }${config.base || "/"}`
           ),
-          __HMR_BASE__: JSON.stringify(server.base ?? "/"),
-          __HMR_TIMEOUT__: JSON.stringify(hmr?.timeout || 30000),
-          __HMR_ENABLE_OVERLAY__: JSON.stringify(hmr?.overlay !== false),
+          __HMR_BASE__: JSON.stringify(serverOptions.base ?? "/"),
+          __HMR_TIMEOUT__: JSON.stringify(hmrOptions?.timeout || 30000),
+          __HMR_ENABLE_OVERLAY__: JSON.stringify(hmrOptions?.overlay !== false),
         },
       };
-      return mergeConfig(config, hmrConfig);
+      return vite.mergeConfig(config, hmrConfig);
     },
-    transform(code, id) {
+    async transform(code, id) {
       // Only transform HTML inputs
       if (!id.endsWith(".html") || !inputIds.includes(id)) return;
 
       const baseUrl = "http://localhost:5173";
 
       // Load scripts from dev server
-      const { document } = parseHTML(code);
+      const { document } = parseHTML(await server.transformIndexHtml(id, code));
```

Comment:
This change right here is does the magic. Everything else is just changing parameters to pass the dev server into the `hmr-rewrite-plugin`